src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -868,33 +897,153 @@ bool HMICapabilitiesImpl::rc_supported() const {
+ */
+bool JsonIsMemberSafe(const Json::Value& json_member,
+                      const char* name_of_member) {
+  return !json_member.isNull() && json_member.isMember(name_of_member);
+}
+
+/**
+ * @brief Converts specified string to appropriate enum value
+ * according to schema
+ * @return converted enum value
+ */
+template <typename EnumType>
+EnumType ConvertStringToEnum(const std::string& str) {
+  using ns_smart_device_link::ns_smart_objects::EnumConversionHelper;
+  EnumType value;
+  if (EnumConversionHelper<EnumType>::StringToEnum(str, &value)) {
+    return value;
+  }
+
+  return EnumType::INVALID_ENUM;
+}
+
+/**
+ * @brief Converts the JSON array of string type into the SmartArray of enums
+ * @param json_array JSON value containing array
+ * @param out_so_array output SmartArray
+ */
+template <typename EnumType>
+void ConvertJsonArrayToSoArray(const Json::Value& json_array,
+                               smart_objects::SmartObject& out_so_array) {
+  out_so_array =
+      smart_objects::SmartObject(smart_objects::SmartType::SmartType_Array);
+  for (uint32_t i = 0; i < json_array.size(); ++i) {
+    out_so_array[i] = ConvertStringToEnum<EnumType>(json_array[i].asString());
+  }
+}
+
+/**
+ * @brief Helper class for obtaining proper capabilities JSON value considering
+ * case if it were overriden by cache
+ */
+struct JsonCapabilitiesGetter {
+ public:
+  /**
+   * @brief JsonCapabilitiesGetter constructor
+   * @param json_default_node reference to the main JSON capabilities node
+   * @param json_cache_node reference to cached JSON capabilities node
+   */
+  JsonCapabilitiesGetter(Json::Value& json_default_node,
+                         Json::Value& json_cache_node)
+      : json_default_node_(json_default_node)
+      , json_cache_node_(json_cache_node) {}
+
+  Json::Value GetInterfaceJsonMember(const char* member_name) {
+    if (JsonIsMemberSafe(json_cache_node_, member_name)) {
+      return GetCachedJsonMember(member_name);
+    }
+
+    if (JsonIsMemberSafe(json_default_node_, member_name)) {
+      return GetMainJsonMember(member_name);
+    }
+
+    return Json::Value::null;
+  }
+
+  /**
+   * @brief GetJsonMember gets JSON value for a specified JSON member from
+   * cached JSON node if member exists, otherwise main JSON node will be
+   * taken
+   * @param member_name name of the JSON member to get
+   * @return JSON value for specified member or Value::null if not found
+   */
+  Json::Value GetJsonMember(
+      const char* member_name,
+      hmi_apis::FunctionID::eType request_id,
+      std::set<hmi_apis::FunctionID::eType>& default_initialized_capabilities) {
+    if (JsonIsMemberSafe(json_cache_node_, member_name)) {
+      return GetCachedJsonMember(member_name);
+    }
+
+    if (JsonIsMemberSafe(json_default_node_, member_name)) {
+      default_initialized_capabilities.insert(request_id);
+      return GetMainJsonMember(member_name);
+    }
+
+    return Json::Value::null;
```

Comment:
@LitvinenkoIra add request_is to the set in case the member_name doesn't present in the default also

Reply:
Done in 
43072e8